_rules/menuitem-non-empty-name-m6b1q3.md

```diff
@@ -0,0 +1,161 @@
+---
+id: m6b1q3
+name: Menuitem has non-empty accessible name
+rule_type: atomic
+description: |
+  This rule checks that each element with a `menuitem` role has a non-empty accessible name.
+accessibility_requirements:
+  wcag20:4.1.2: # Name, Role, Value (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility Tree
+  - CSS Styling
+  - DOM Tree
+acknowledgments:
+  authors:
+    - Wilco Fiers
+  assets:
+    - The "file.svg" icon is part of the Feather Icons package, which uses the [MIT license](https://github.com/feathericons/feather/blob/master/LICENSE).
+---
+
+## Applicability
+
+This rule applies to HTML elements with a `menuitem` [semantic role][] that are [included in the accessibility tree][].
+
+## Expectation
+
+Each target element has an [accessible name][] that is not empty (`""`).
+
+## Assumptions
+
+This rule assumes that all menuitems are [user interface components as defined by WCAG 2](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). If an element has a role of `menuitem` that would not be perceived as a single control by users, [4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value) would not apply and so failing this rule would not result in a conformance issue.
+
+## Accessibility Support
+
+Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some elements can have a [semantic role][] of `button` and fail this rule with some technology but users of other technologies would not experience any accessibility issue.
+
```

Comment:
Would it be worth to mention the support issues for the role of `menuitem`? I've done a quick test using JAWS and NVDA on FF and Chrome. NVDA only announces menu items in the browsing mode but not when they receive keyboard focus. Also, NVDA seems to support the `title` attribute as accessible name only in the browsing mode.